### PR TITLE
Fixed #25663 -- Added checking of the number of points for `LinearRing` and `LineString`.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/25663

### CODE REVIEW

File: tests/gis_tests/geos_tests/test_geos.py [hunk 1]
```diff
@@ -28,7 +27,7 @@
 
 
 @skipUnless(HAS_GEOS, "Geos is required.")
-class GEOSTest(unittest.TestCase, TestDataMixin):
+class GEOSTest(TestCase, TestDataMixin):
```
Comment (charettes) [inline tests/gis_tests/geos_tests/test_geos.py]: Use a `SimpleTestCase` instead as it provides `assertRaisesMessage` and avoids wrapping each test case in a transaction which is not required here.

Comment (sir-sigurd) [inline tests/gis_tests/geos_tests/test_geos.py]: Fixed, thanks for the note.

File: django/contrib/gis/geos/linestring.py [hunk 2]
```diff
@@ -36,16 +36,20 @@ def __init__(self, *args, **kwargs):
...
             #  must stay the same, e.g., no LineString((1, 2), (1, 2, 3)).
             ncoords = len(coords)
-            if coords:
-                ndim = len(coords[0])
-            else:
-                raise TypeError('Cannot initialize on empty sequence.')
-            self._checkdim(ndim)
+            if ncoords < self._minlength:
+                raise TypeError(
+                    '`%s` requires at least %d point, got %s.' % (self.__class__.__name__, self._minlength, ncoords)
```
Comment (claudep) [inline django/contrib/gis/geos/linestring.py]: point -> points

**Comment by timgraham** [comment]
merged in 1e35dd1a053b620f065101c02069c8e142ebf8ec, thanks!

For future reference, in Django I haven't seen your convention of including backticks around class names in commit messages, exception messages, etc, so I suggest to be consistent with that.

**Pull request closed without merging**